src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Bindings/WatchableCloudBlobStream.cs

```diff
@@ -153,5 +191,20 @@ public ParameterLog GetStatus()
                 return null;
             }
         }
+
+        private void Log()
+        {
+            if (_logged || _logger == null || _blob == null)
+            {
+                return;
+            }
+
+            // The ETag may have been created/updated after the Stream is committed, so we read it now
+            _blobAccessLog["ETag"] = _blob.Properties.ETag;
```

Comment:
Rather than using json serialization for this log, for performance, let's follow the pattern we've used elsewhere and use a compiled log statement via LoggerMessage.Define, e.g. as we do in the host [here](https://github.com/Azure/azure-functions-host/blob/f9053547403d17e0aae154db3db39f809e5817b3/src/WebJobs.Script.WebHost/Middleware/SystemTraceMiddleware.cs#L34). This saves us a few allocations, and some json serialization cost.